### DISCUSSION TITLE
<Link> to external page

### DISCUSSION THREAD

**Discussion by Manubi** [remix-run/react-router#8611] [Q&A]
Hi,
I can't find anything in the documentation about how to use `<Link />` with external urls. Just found this github issue... 
https://github.com/remix-run/react-router/issues/1147

Is it still the case that I need to implement the check myself if it's a `App link` or `External page link`? 

I just want to implement a list of urls that are either on our app or link to an external page. 

btw the [react-router-hash-link](https://github.com/rafgraph/react-router-hash-link) is also not implemented?

btw2: The documentation of the router v6 without a search is rather difficult to navigate. 

Thanks!

**Comment by brianespinosa** [comment]
[This is still the best answer to external links](https://github.com/remix-run/react-router/issues/1147#issuecomment-113180174).

Developers should be aware that their links outside of the application have nothing to do with the internal router at all and should reach for the appropriate tag versus using a Link component. You wanted to make sure to always use the came component everywhere, you could write your own that does the check to determine if it is external or internal, and even add some of the safe checks for `noopener noferrer` if you are opening a `target="_blank"`. An example of how you might be able to build that component is also in the issue you reference.

**Reply by Manubi** [reply]
Thanks, I found that thread. But looking at the feedback I imagined the check would've been implemented by now into the router. Of course option in. But yeah... 
![CleanShot 2022-01-25 at 11 28 01](https://user-images.githubusercontent.com/20935138/150959765-64cc48bb-54d4-4f60-bb49-ae0bdafe3fba.png)

Imho it doesn't help a lot to keep telling people that it has _"nothing to do with the internal router"_. People make an array of objects with internal and external urls. People map over these objects. So in my view it would make sense to listen to people and make the DX better. Same with the hash-link mentioned above.

Anyway. Wish you a nice day. 
 
![CleanShot 2022-01-25 at 11 38 59](https://user-images.githubusercontent.com/20935138/150961494-b8278343-b55c-41cd-84c9-e3dd36957962.png)

And something I just found out **now**:
![CleanShot 2022-01-25 at 14 34 42](https://user-images.githubusercontent.com/20935138/150986749-64fecef9-b640-4418-a68b-8d9ceeda35ba.png)

**Reply by brianespinosa** [reply]
The goal in pointing that out is to make sure there is an understanding about how something works so we don't just blindly feature request things that might semantically be beyond the scope of a library. Once that happens, if there is magic happening where the library checks things outside of the router's domain, where do the requests stop? That is why I referenced an example talking about `target` and `rel` attributes.

**Reply by Manubi** [reply]
The point is, almost everyone has to deal with external links in an app. I always thought it's common sense to improve the DX but I guess some artificial rules made up by someone are more important. Anyway, gl and hf. 😄

**Reply by brianespinosa** [reply]
Bud... I don't even maintain RR. I am just pointing you to the still current best answer to your question. Sorry you don't like the answer the maintainers are giving and the reasoning behind it.

**Reply by Manubi** [reply]
Thanks. What do u think?😅

**Comment by mfpopa** [comment]
`<Link />` is meant to be used when you want to change the application route, which is an internal link inside your application, not to navigate to external links. External links will never be supported by React Router via the `<Link />` component for that simple reason.

If you need a link component that has the same branding regardless if it's an internal or external link, then you should create one component that can differentiate between internal and external links as suggested by Ryan Florence and add your own branding. Or just create two separate components with your branding, one for external links, one for internal links that wraps around the React Router `<Link />` component.

**Comment by ChristopherChudzicki** [comment]
I believe the above comments are outdated—This seems to have changed in January 2023 in https://github.com/remix-run/react-router/pull/9900

`Link` behaves as one would expect for both internal and external links:
- `<Link to="/search">hello</Link>` is an in-app link that does not trigger a page reload
- ``<Link to={`${window.origin}/search`}>hello</Link>`` is an in-app link that does not trigger a page reload
- `<Link to="https://github.com">hello</Link>` triggers external navigation to github.

The exact logic can be seen here: https://github.com/remix-run/react-router/blob/94f7fb7b3a3d2ef266f1d83293c5ab50c1963132/packages/react-router-dom/index.tsx#L957-L971

I've failed to find this mentioned anywhere in the docs, though it is consistent with @mjackson's comment in https://github.com/remix-run/react-router/issues/1147#issuecomment-915503932
